[BUG] `'Package' does not contain a definition for 'targetFramework'`

**Describe the bug**
Well. A bug
```
Library\PackageCache\org.kumas.nuget-importer@e006d74628\Editor\PackageOperation\OperatePackage.cs(251,112): error CS1061: 'Package' does not contain a definition for 'targetFramework' and no accessible extension method 'targetFramework' accepting a first argument of type 'Package' could be found (are you missing a using directive or an assembly reference?)
```

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/FrameworkName.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/FrameworkName.cs
@@ -140,15 +140,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                 List<string> ret;
                 switch (PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup))
                 {
-                    case ApiCompatibilityLevel.NET_4_6:
-                        ret = NET4_6;
-                        ret.AddRange(STANDARD2_0);
-                        break;
-                    case ApiCompatibilityLevel.NET_Standard_2_0:
-                        ret = STANDARD2_0;
-                        ret.AddRange(NET4_6);
-                        break;
-#if UNITY_2021_2_1_OR_NEWER
+#if UNITY_2021_2_OR_NEWER
                     case ApiCompatibilityLevel.NET_Unity_4_8:
                         ret = NET4_8;
                         ret.AddRange(STANDARD2_1);
@@ -156,6 +148,15 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                     case ApiCompatibilityLevel.NET_Standard:
                         ret = STANDARD2_1;
                         ret.AddRange(NET4_8);
+                        break;
+#else
+                    case ApiCompatibilityLevel.NET_4_6:
+                        ret = NET4_6;
+                        ret.AddRange(STANDARD2_0);
+                        break;
+                    case ApiCompatibilityLevel.NET_Standard_2_0:
+                        ret = STANDARD2_0;
+                        ret.AddRange(NET4_6);
                         break;
 #endif
                     case ApiCompatibilityLevel.NET_2_0:

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
@@ -341,7 +341,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                 var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
                 var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
-                var stringVersion =  version.Aggregate((prev, curr) => $"{prev}.{curr}");
+                var stringVersion = version.Aggregate((prev, curr) => $"{prev}.{curr}");
                 if (maxDiff == 0 && !_excludeMaximum)
                 {
                     return minDiff == 0 && !_excludeMinimum
@@ -398,7 +398,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                 var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                 var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
-                var stringVersion =  version.Aggregate((prev, curr) => $"{prev}.{curr}");
+                var stringVersion = version.Aggregate((prev, curr) => $"{prev}.{curr}");
                 if (minDiff == 0 && !_excludeMinimum)
                 {
                     return maxDiff == 0 && !_excludeMaximum
@@ -565,8 +565,16 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                 switch (isNumberCompare)
                 {
-                    case true when isNumberBase && c != b: return c - b;
-                    case false when isNumberBase == false:
+                    case true when isNumberBase:
+                        {
+                            if (c != b)
+                            {
+                                return c - b;
+                            }
+
+                            break;
+                        }
+                    case false when !isNumberBase:
                         {
                             var compare = string.CompareOrdinal(splitedCompare[0], splitedBase[0]);
                             if (compare != 0)

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
@@ -341,7 +341,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                 var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
                 var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
-                var stringVersion = string.Join(".", version);
+                var stringVersion =  version.Aggregate((prev, curr) => $"{prev}.{curr}");
                 if (maxDiff == 0 && !_excludeMaximum)
                 {
                     return minDiff == 0 && !_excludeMinimum
@@ -349,7 +349,12 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                         : minDiff > 0
                             ? stringVersion
                             : throw new InvalidOperationException(
-                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                $@"There is no available version. Version: {stringVersion} Min: {
+                                    _minimumVersion?.Aggregate(
+                                        (prev, curr) => $"{prev}.{curr}"
+                                    )} Max: {_maximumVersion?.Aggregate(
+                                    (prev, curr) => $"{prev}.{curr}"
+                                )}"
                             );
                 }
 
@@ -360,13 +365,23 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                         : minDiff > 0
                             ? stringVersion
                             : throw new InvalidOperationException(
-                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                $@"There is no available version. Version: {stringVersion} Min: {
+                                    _minimumVersion?.Aggregate(
+                                        (prev, curr) => $"{prev}.{curr}"
+                                    )} Max: {_maximumVersion?.Aggregate(
+                                    (prev, curr) => $"{prev}.{curr}"
+                                )}"
                             );
                 }
             }
 
+
             throw new InvalidOperationException(
-                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+                $@"There is no available version. Versions: {
+                    _existVersions.Select(it => it?.Aggregate((prev, curr) => $"{prev}.{curr}"))
+                                  .Aggregate((prev, curr) => $"{prev}, {curr}")} Min: {
+                                  _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                      _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
             );
         }
 
@@ -383,7 +398,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                 var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                 var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
-                var stringVersion = string.Join(".", version);
+                var stringVersion =  version.Aggregate((prev, curr) => $"{prev}.{curr}");
                 if (minDiff == 0 && !_excludeMinimum)
                 {
                     return maxDiff == 0 && !_excludeMaximum
@@ -391,7 +406,12 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                         : maxDiff < 0
                             ? stringVersion
                             : throw new InvalidOperationException(
-                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                $@"There is no available version. Version: {stringVersion} Min: {
+                                    _minimumVersion?.Aggregate(
+                                        (prev, curr) => $"{prev}.{curr}"
+                                    )} Max: {_maximumVersion?.Aggregate(
+                                    (prev, curr) => $"{prev}.{curr}"
+                                )}"
                             );
                 }
 
@@ -402,13 +422,23 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                         : maxDiff < 0
                             ? stringVersion
                             : throw new InvalidOperationException(
-                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                $@"There is no available version. Version: {stringVersion} Min: {
+                                    _minimumVersion?.Aggregate(
+                                        (prev, curr) => $"{prev}.{curr}"
+                                    )} Max: {_maximumVersion?.Aggregate(
+                                    (prev, curr) => $"{prev}.{curr}"
+                                )}"
                             );
                 }
             }
 
+
             throw new InvalidOperationException(
-                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+                $@"There is no available version. Versions: {
+                    _existVersions?.Select(it => it?.Aggregate((prev, curr) => $"{prev}.{curr}"))
+                                  .Aggregate((prev, curr) => $"{prev}, {curr}")} Min: {
+                                  _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                      _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
             );
         }
 
@@ -427,7 +457,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
 
                     var maxDiff = CompareVersion(version, _maximumVersion);
                     var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
-                    var stringVersion = string.Join(".", version);
+                    var stringVersion = version.Aggregate((prev, curr) => $"{prev}.{curr}");
                     if (maxDiff == 0 && !_excludeMaximum)
                     {
                         return minDiff == 0 && !_excludeMinimum
@@ -435,7 +465,9 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                             : minDiff > 0
                                 ? stringVersion
                                 : throw new InvalidOperationException(
-                                    $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                    $@"There is no available version. Version: {stringVersion} Min: {
+                                        _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                            _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
                                 );
                     }
 
@@ -446,14 +478,20 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                             : minDiff > 0
                                 ? stringVersion
                                 : throw new InvalidOperationException(
-                                    $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                    $@"There is no available version. Version: {stringVersion} Min: {
+                                        _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                            _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
                                 );
                     }
                 }
 
 
                 throw new InvalidOperationException(
-                    $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+                    $@"There is no available version. Versions: {
+                        _existVersions.Select(it => it?.Aggregate((prev, curr) => $"{prev}.{curr}"))
+                                      .Aggregate((prev, curr) => $"{prev}, {curr}")} Min: {
+                                      _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                          _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
                 );
             }
 
@@ -479,7 +517,11 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             }
 
             throw new InvalidOperationException(
-                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+                $@"There is no available version. Versions: {
+                    _existVersions.Select(it => it?.Aggregate((prev, curr) => $"{prev}.{curr}"))
+                                  .Aggregate((prev, curr) => $"{prev}, {curr}")} Min: {
+                                  _minimumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")} Max: {
+                                      _maximumVersion?.Aggregate((prev, curr) => $"{prev}.{curr}")}"
             );
         }
 

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
@@ -104,8 +104,10 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             {
                 _allowedVersion = value;
                 var splitedNotion = value.Replace(" ", "").Split(',');
-                switch (splitedNotion.Length) {
-                    case 1: {
+                switch (splitedNotion.Length)
+                {
+                    case 1:
+                        {
                             _excludeMinimum = false;
                             if (value.StartsWith("["))
                             {
@@ -198,7 +200,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
         /// <para>安定版のみつかうか。</para>
         /// </param>
         /// <returns>
-        /// <para>Marged Semantic Versioning.</para>
+        /// <para>Merged Semantic Versioning.</para>
         /// <para>マージされたセマンティックバージョニング。</para>
         /// </returns>
         /// <exception cref="System.ArgumentException">
@@ -210,7 +212,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             var ret = new SemVer();
             if (!ExistVersion.SequenceEqual(newVersion.ExistVersion))
             {
-                throw new ArgumentException("These version can't marge. Because exist versions are different.");
+                throw new ArgumentException("These version can't merge. Because exist versions are different.");
             }
 
             ret._selectedVersion = _selectedVersion;
@@ -253,9 +255,9 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             {
                 ret.GetSuitVersion(onlyStable);
             }
-            catch (InvalidOperationException)
+            catch (InvalidOperationException e)
             {
-                throw new ArgumentException("These version can't marge. Because there is no suitable version.");
+                throw new ArgumentException("These version can't merge. Because there is no suitable version.", e);
             }
 
             var allowVer = "";
@@ -330,11 +332,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
         private string GetHighestVersion(bool onlyStable)
         {
             List<string> maxVersion = _maximumVersion ?? _existVersions.First();
-            foreach (List<string> version in _existVersions.SkipWhile(
-                                                               ver => ver.Aggregate((now, next) => now + next)
-                                                                      != maxVersion.Aggregate((now, next) => now + next)
-                                                           )
-                                                           .ToArray())
+            foreach (List<string> version in _existVersions.SkipWhile(ver => ver.SequenceEqual(maxVersion)))
             {
                 if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
                 {
@@ -342,68 +340,76 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                 }
 
                 var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
+                var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
+                var stringVersion = string.Join(".", version);
                 if (maxDiff == 0 && !_excludeMaximum)
                 {
-                    var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                     return minDiff == 0 && !_excludeMinimum
-                        ? version.Aggregate((now, next) => now + "." + next)
+                        ? stringVersion
                         : minDiff > 0
-                            ? version.Aggregate((now, next) => now + "." + next)
-                            : throw new InvalidOperationException("There is no available version.");
+                            ? stringVersion
+                            : throw new InvalidOperationException(
+                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                            );
                 }
 
                 if (maxDiff < 0)
                 {
-                    var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                     return minDiff == 0 && !_excludeMinimum
-                        ? version.Aggregate((now, next) => now + "." + next)
+                        ? stringVersion
                         : minDiff > 0
-                            ? version.Aggregate((now, next) => now + "." + next)
-                            : throw new InvalidOperationException("There is no available version.");
+                            ? stringVersion
+                            : throw new InvalidOperationException(
+                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                            );
                 }
             }
 
-            throw new InvalidOperationException("There is no available version.");
+            throw new InvalidOperationException(
+                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+            );
         }
 
         private string GetLowestVersion(bool onlyStable)
         {
             List<string> minVersion = _minimumVersion ?? _existVersions.Last();
-            foreach (List<string> version in (_existVersions.AsEnumerable() ?? Array.Empty<List<string>>()).Reverse()
-                     .SkipWhile(
-                         ver => ver.Aggregate((now, next) => now + next)
-                                != minVersion.Aggregate((now, next) => now + next)
-                     )
-                     .ToArray())
+            foreach (List<string> version in _existVersions.AsEnumerable()!.Reverse()
+                                                           .SkipWhile(ver => ver.SequenceEqual(minVersion)))
             {
                 if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
                 {
                     continue;
                 }
 
-                var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 11;
+                var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
+                var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
+                var stringVersion = string.Join(".", version);
                 if (minDiff == 0 && !_excludeMinimum)
                 {
-                    var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
                     return maxDiff == 0 && !_excludeMaximum
-                        ? version.Aggregate((now, next) => now + "." + next)
+                        ? stringVersion
                         : maxDiff < 0
-                            ? version.Aggregate((now, next) => now + "." + next)
-                            : throw new InvalidOperationException("There is no available version.");
+                            ? stringVersion
+                            : throw new InvalidOperationException(
+                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                            );
                 }
 
                 if (minDiff > 0)
                 {
-                    var maxDiff = _maximumVersion != null ? CompareVersion(version, _maximumVersion) : -1;
                     return maxDiff == 0 && !_excludeMaximum
-                        ? version.Aggregate((now, next) => now + "." + next)
+                        ? stringVersion
                         : maxDiff < 0
-                            ? version.Aggregate((now, next) => now + "." + next)
-                            : throw new InvalidOperationException("There is no available version.");
+                            ? stringVersion
+                            : throw new InvalidOperationException(
+                                $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                            );
                 }
             }
 
-            throw new InvalidOperationException("There is no available version.");
+            throw new InvalidOperationException(
+                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+            );
         }
 
         private string GetSuitVersion(bool onlyStable)
@@ -411,12 +417,8 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             if (_maximumVersion != null)
             {
                 foreach (List<string> version in _existVersions.SkipWhile(
-                                                                   ver => ver.Aggregate((now, next) => now + next)
-                                                                          != _maximumVersion.Aggregate(
-                                                                              (now, next) => now + next
-                                                                          )
-                                                               )
-                                                               .ToArray())
+                             ver => ver.SequenceEqual(_maximumVersion)
+                         ))
                 {
                     if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
                     {
@@ -424,58 +426,61 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
                     }
 
                     var maxDiff = CompareVersion(version, _maximumVersion);
+                    var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
+                    var stringVersion = string.Join(".", version);
                     if (maxDiff == 0 && !_excludeMaximum)
                     {
-                        var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                         return minDiff == 0 && !_excludeMinimum
-                            ? version.Aggregate((now, next) => now + "." + next)
+                            ? stringVersion
                             : minDiff > 0
-                                ? version.Aggregate((now, next) => now + "." + next)
-                                : throw new InvalidOperationException("There is no available version.");
+                                ? stringVersion
+                                : throw new InvalidOperationException(
+                                    $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                );
                     }
 
                     if (maxDiff < 0)
                     {
-                        var minDiff = _minimumVersion != null ? CompareVersion(version, _minimumVersion) : 1;
                         return minDiff == 0 && !_excludeMinimum
-                            ? version.Aggregate((now, next) => now + "." + next)
+                            ? stringVersion
                             : minDiff > 0
-                                ? version.Aggregate((now, next) => now + "." + next)
-                                : throw new InvalidOperationException("There is no available version.");
+                                ? stringVersion
+                                : throw new InvalidOperationException(
+                                    $"There is no available version. Version: {stringVersion} Min: {_minimumVersion} Max: {_maximumVersion}"
+                                );
                     }
                 }
 
-                throw new InvalidOperationException("There is no available version.");
+
+                throw new InvalidOperationException(
+                    $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+                );
             }
 
             var highList = new List<List<string>>();
             var majorVersion = new List<string>();
 
-            foreach (List<string> version in _existVersions)
+            foreach (var version in _existVersions
+                                    .Where(
+                                        version => !onlyStable
+                                                   || (!version.Any(v => v.Contains('-')) && version[0][0] != '0')
+                                    )
+                                    .Where(version => !majorVersion.Contains(version[0])))
             {
-                if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
-                {
-                    continue;
-                }
-
-                if (!majorVersion.Contains(version[0]))
-                {
-                    highList.Add(version);
-                    majorVersion.Add(version[0]);
-                }
+                highList.Add(version);
+                majorVersion.Add(version[0]);
             }
 
             highList.Reverse();
 
-            foreach (List<string> high in highList)
+            foreach (var high in highList.Where(IsAllowedVersion))
             {
-                if (IsAllowedVersion(high))
-                {
-                    return high.Aggregate((now, next) => now + "." + next);
-                }
+                return high.Aggregate((now, next) => now + "." + next);
             }
 
-            throw new InvalidOperationException("There is no available version.");
+            throw new InvalidOperationException(
+                $"There is no available version. Versions: {_existVersions} Min: {_minimumVersion} Max: {_maximumVersion}"
+            );
         }
 
         /// <summary>
@@ -506,7 +511,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
         /// <param name="compareVersion">compare version</param>
         /// <param name="baseVersion">base version</param>
         /// <returns>If compareVersion is newer version than baseVersion, return positive number. If baseVersion is newer version than compareVersion, return negative number. If both are same version, return 0.</returns>
-        private static int CompareVersion(List<string> compareVersion, List<string> baseVersion)
+        private static int CompareVersion(IReadOnlyList<string> compareVersion, IReadOnlyList<string> baseVersion)
         {
             var minLength = compareVersion.Count < baseVersion.Count ? compareVersion.Count : baseVersion.Count;
             for (var i = 0; i < minLength; i++)

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/DataClasses/SemVer.cs
@@ -332,7 +332,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
         private string GetHighestVersion(bool onlyStable)
         {
             List<string> maxVersion = _maximumVersion ?? _existVersions.First();
-            foreach (List<string> version in _existVersions.SkipWhile(ver => ver.SequenceEqual(maxVersion)))
+            foreach (List<string> version in _existVersions.SkipWhile(ver => !ver.SequenceEqual(maxVersion)))
             {
                 if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
                 {
@@ -389,7 +389,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
         {
             List<string> minVersion = _minimumVersion ?? _existVersions.Last();
             foreach (List<string> version in _existVersions.AsEnumerable()!.Reverse()
-                                                           .SkipWhile(ver => ver.SequenceEqual(minVersion)))
+                                                           .SkipWhile(ver => !ver.SequenceEqual(minVersion)))
             {
                 if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))
                 {
@@ -447,7 +447,7 @@ namespace kumaS.NuGetImporter.Editor.DataClasses
             if (_maximumVersion != null)
             {
                 foreach (List<string> version in _existVersions.SkipWhile(
-                             ver => ver.SequenceEqual(_maximumVersion)
+                             ver => !ver.SequenceEqual(_maximumVersion)
                          ))
                 {
                     if (onlyStable && (version.Any(v => v.Contains('-')) || version[0][0] == '0'))

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterSettings.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterSettings.cs
@@ -41,6 +41,7 @@ namespace kumaS.NuGetImporter.Editor
                     return _instance;
                 }
 
+                EnsureSetProjectSettingsPath();
                 var path = Path.Combine(_projectSettingsPath, "NuGetImporterSettings.json");
                 if (!File.Exists(path))
                 {

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterWindow.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterWindow.cs
@@ -174,12 +174,8 @@ namespace kumaS.NuGetImporter.Editor
             }
 
             var text = await response.Content.ReadAsStringAsync();
-            MatchCollection matchs = Regex.Matches(text, @"NuGetImporterForUnity\.(?<version>\d+\.\d+\.\d+)\.zip");
-            var versions = new List<string>();
-            foreach (Match match in matchs)
-            {
-                versions.Add(match.Groups["version"].Value);
-            }
+            var matches = Regex.Matches(text, @"NuGetImporterForUnity\.(?<version>\d+\.\d+\.\d+)\.zip");
+            var versions = (from Match match in matches select match.Groups["version"].Value).ToList();
 
             var latestVersion = SemVer.SortVersion(versions)[0];
             Version thisVersion = typeof(NuGetImporterWindow).Assembly.GetName().Version;
@@ -356,7 +352,8 @@ namespace kumaS.NuGetImporter.Editor
             }
 
             var package =
-                PackageManager.Installed.Package.Where(package => package.ID == data.items[0].items[0].catalogEntry.id).ToList();
+                PackageManager.Installed.Package.Where(package => package.ID == data.items[0].items[0].catalogEntry.id)
+                              .ToArray();
             if (package.Any())
             {
                 _summary = new PackageSummary(data, package.First().Version);
@@ -480,8 +477,10 @@ namespace kumaS.NuGetImporter.Editor
                     {
                         _catalogs.Add(catalog.Value);
                         var package = PackageManager.Installed.Package.Where(
-                            package => package.ID == catalog.Value.items[0].items[0].catalogEntry.id
-                        ).ToList();
+                                                        package => package.ID
+                                                                   == catalog.Value.items[0].items[0].catalogEntry.id
+                                                    )
+                                                    .ToList();
                         if (package.Any())
                         {
                             tasks.Add(catalog.Value.GetIcon(package.First().Version));

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageManager.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageManager.cs
@@ -776,7 +776,7 @@ namespace kumaS.NuGetImporter.Editor
             }
 
             installed.Package.AddRange(result.Where(value => !value.isSkipped).Select(value => value.package));
-            existingPackage.Package.AddRange(ret.Select(r => new Package() { ID = r.ID, Version = "0.0.0" }));
+            existingPackage.Package.AddRange(ret.Select(r => new Package { ID = r.ID, Version = "0.0.0" }));
 
             return ret;
         }

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/ConvertToUPM.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/ConvertToUPM.cs
@@ -65,9 +65,7 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 EditorUtility.DisplayDialog(
                     "NuGet importer",
                     "Error occured!\nRolls back to before the operation.\nError :\n"
-                    + install.Exception!.Message
-                    + "\n"
-                    + install.Exception!.StackTrace,
+                    + install.Exception!.Message,
                     "OK"
                 );
                 await Rollback(controlledPackages, operateLock, controller);

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/ConvertToUPM.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/ConvertToUPM.cs
@@ -46,7 +46,7 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 );
                 var install = PackageManager.InstallSelectPackages(
                     controlledPackages.Installed,
-                    new string[0],
+                    Array.Empty<string>(),
                     operateLock,
                     new PackageControllerAsUPM()
                 );
@@ -56,19 +56,22 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                     controlledPackages.Installed.Select(package => package.ID).ToArray()
                 );
                 await install;
-                if (install.IsFaulted)
+                if (!install.IsFaulted)
                 {
-                    UnityEngine.Debug.LogException(install.Exception);
-                    EditorUtility.DisplayDialog(
-                        "NuGet importer",
-                        "Error occured!\nRolls back to before the operation.\nError :\n" + install.Exception!.Message,
-                        "OK"
-                    );
-                    await Rollback(controlledPackages, operateLock, controller);
-                    return new OperationResult(OperationState.Failure, "Rollback to before operation is complete.");
+                    return new OperationResult(OperationState.Success, FinishMessage);
                 }
 
-                return new OperationResult(OperationState.Success, FinishMessage);
+                UnityEngine.Debug.LogException(install.Exception);
+                EditorUtility.DisplayDialog(
+                    "NuGet importer",
+                    "Error occured!\nRolls back to before the operation.\nError :\n"
+                    + install.Exception!.Message
+                    + "\n"
+                    + install.Exception!.StackTrace,
+                    "OK"
+                );
+                await Rollback(controlledPackages, operateLock, controller);
+                return new OperationResult(OperationState.Failure, "Rollback to before operation is complete.");
             }
 
             EditorUtility.DisplayDialog(

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/OperatePackage.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/OperatePackage.cs
@@ -58,14 +58,21 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 EditorUtility.DisplayProgressBar("NuGet importer", "Solving dependency", 0);
                 OperationResult result = await Operate(controlledPackages, operatorLock);
                 ret = result.State;
-                await AsmdefController.UpdateAsmdef(PackageManager.ControlledPackages.Installed, PackageManager.GetPackagePathSolver());
+                await AsmdefController.UpdateAsmdef(
+                    PackageManager.ControlledPackages.Installed,
+                    PackageManager.GetPackagePathSolver()
+                );
                 return result;
             }
             catch (Exception e)
             {
                 ret = OperationState.Failure;
                 UnityEngine.Debug.LogException(e);
-                EditorUtility.DisplayDialog("NuGet importer", "Error occured!\nRolls back to before the operation.\nError :\n" + e.Message, "OK");
+                EditorUtility.DisplayDialog(
+                    "NuGet importer",
+                    "Error occured!\nRolls back to before the operation.\nError :\n" + e.Message + "\n" + e.StackTrace,
+                    "OK"
+                );
                 await Rollback(controlledPackages, operatorLock);
                 return new OperationResult(ret, "Rollback to before operation is complete.");
             }
@@ -108,14 +115,32 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
         /// <para>Manipulate result.</para>
         /// <para>操作結果。</para>
         /// </returns>
-        protected async Task<OperationResult> ManipulatePackages(ICollection<Package> rootPackages, ICollection<Package> installPackages, ICollection<Package> deletePackages, ReadOnlyControlledPackages controlledPackages, PackageManager.OperateLock operateLock, PackageControllerBase controller = default)
+        protected async Task<OperationResult> ManipulatePackages(
+            ICollection<Package> rootPackages,
+            ICollection<Package> installPackages,
+            ICollection<Package> deletePackages,
+            ReadOnlyControlledPackages controlledPackages,
+            PackageManager.OperateLock operateLock,
+            PackageControllerBase controller = default
+        )
         {
             OperationState ret;
 
             InstallingPackages.AddRange(installPackages);
-            Package[] uninstallPackages = deletePackages.Where(package => installPackages.All(install => install.ID != package.ID)).ToArray();
-            Package[] changePackages = deletePackages.Where(package => installPackages.Any(install => install.ID == package.ID)).ToArray();
-            Package[] installedPackages = controlledPackages.Installed.Where(package => deletePackages.All(uninstall => uninstall.ID != package.ID)).Concat(installPackages).ToArray();
+            Package[] uninstallPackages = deletePackages
+                                          .Where(package => installPackages.All(install => install.ID != package.ID))
+                                          .ToArray();
+            Package[] changePackages = deletePackages
+                                       .Where(package => installPackages.Any(install => install.ID == package.ID))
+                                       .ToArray();
+            Package[] installedPackages = controlledPackages.Installed
+                                                            .Where(
+                                                                package => deletePackages.All(
+                                                                    uninstall => uninstall.ID != package.ID
+                                                                )
+                                                            )
+                                                            .Concat(installPackages)
+                                                            .ToArray();
 
             var nativePackages = new List<Package>();
             var managedPackages = new List<Package>();
@@ -146,7 +171,15 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
 
                 if (nativePackages.Any())
                 {
-                    Process process = await PackageManager.OperateWithNativeAsync(installPackages, managedPackages, nativePackages, installedPackages, rootPackages, operateLock, controller);
+                    Process process = await PackageManager.OperateWithNativeAsync(
+                        installPackages,
+                        managedPackages,
+                        nativePackages,
+                        installedPackages,
+                        rootPackages,
+                        operateLock,
+                        controller
+                    );
                     ret = OperationState.Success;
                     operateLock.result = ret;
                     AssetDatabase.SaveAssets();
@@ -161,8 +194,11 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 }
             }
 
-            IEnumerable<string> loadedAsmNames = AppDomain.CurrentDomain.GetAssemblies().Select(asm => asm.GetName().Name);
-            loadedAsmNames = loadedAsmNames.Except(PackageManager.PackageAsmNames.managedList.SelectMany(pkg => pkg.fileNames)).ToArray();
+            IEnumerable<string> loadedAsmNames =
+                AppDomain.CurrentDomain.GetAssemblies().Select(asm => asm.GetName().Name);
+            loadedAsmNames = loadedAsmNames
+                             .Except(PackageManager.PackageAsmNames.managedList.SelectMany(pkg => pkg.fileNames))
+                             .ToArray();
 
             if (changePackages.Any())
             {
@@ -172,7 +208,12 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
             ICollection<Package> skipped = new List<Package>();
             if (installPackages.Any())
             {
-                Task<ICollection<Package>> tasks = PackageManager.InstallSelectPackages(installPackages, loadedAsmNames, operateLock, controller);
+                Task<ICollection<Package>> tasks = PackageManager.InstallSelectPackages(
+                    installPackages,
+                    loadedAsmNames,
+                    operateLock,
+                    controller
+                );
 
                 _ = PackageManager.DownloadProgress(0.5f, installPackages.Select(package => package.ID).ToArray());
                 skipped = await tasks;
@@ -180,7 +221,11 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 {
                     ret = OperationState.Failure;
                     UnityEngine.Debug.LogException(tasks.Exception);
-                    EditorUtility.DisplayDialog("NuGet importer", "Error occured!\nRolls back to before the operation.\nError :\n" + tasks.Exception!.Message, "OK");
+                    EditorUtility.DisplayDialog(
+                        "NuGet importer",
+                        "Error occured!\nRolls back to before the operation.\nError :\n" + tasks.Exception!.Message,
+                        "OK"
+                    );
                     await Rollback(controlledPackages, operateLock, controller);
                     return new OperationResult(ret, "Rollback to before operation is complete.");
                 }
@@ -188,8 +233,12 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
 
             if (skipped.Any())
             {
-                EditorUtility.DisplayDialog("NuGet importer", "The below packages are existing in your project so we skipped installing them.\n\n" +
-                    skipped.Select(pkg => pkg.ID).Aggregate((now, next) => now + "\n" + next), "OK");
+                EditorUtility.DisplayDialog(
+                    "NuGet importer",
+                    "The below packages are existing in your project so we skipped installing them.\n\n"
+                    + skipped.Select(pkg => pkg.ID).Aggregate((now, next) => now + "\n" + next),
+                    "OK"
+                );
             }
 
             ret = OperationState.Success;
@@ -216,7 +265,12 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
         /// <para>Return true when the user agrees.</para>
         /// <para>ユーザーが了解したか。</para>
         /// </returns>
-        private static async Task<bool> ConfirmToUser(ICollection<Package> installPackages, ICollection<Package> uninstallPackages, IEnumerable<Package> nativePackages, ReadOnlyControlledPackages controlledPackages)
+        private static async Task<bool> ConfirmToUser(
+            ICollection<Package> installPackages,
+            ICollection<Package> uninstallPackages,
+            IEnumerable<Package> nativePackages,
+            ReadOnlyControlledPackages controlledPackages
+        )
         {
             var builder = new StringBuilder();
             if (uninstallPackages.Any())
@@ -248,19 +302,31 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
 
 #if UNITY_2021_2_OR_NEWER
             if(PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup) == ApiCompatibilityLevel.NET_Standard){
-                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_1.Contains(package.TargetFramework)).ToArray();
+                warningPackages =
+ installPackages.Where(package => !FrameworkName.STANDARD2_1.Contains(package.TargetFramework)).ToArray();
             }
 #else
-            if (PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup) == ApiCompatibilityLevel.NET_Standard_2_0)
+            if (PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup)
+                == ApiCompatibilityLevel.NET_Standard_2_0)
             {
-                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_0.Contains(package.TargetFramework)).ToArray();
+                warningPackages = installPackages
+                                  .Where(package => !FrameworkName.STANDARD2_0.Contains(package.TargetFramework))
+                                  .ToArray();
             }
 #endif
             if (warningPackages.Any())
             {
-                if (!EditorUtility.DisplayDialog("Warning from NuGet importer", "Now the api compatibility level for this project is " +
-                    PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup).ToString() +
-                    ". But below packages are builded for .NETFramework. Do you install them?" + "\n\n" + string.Join("\n", warningPackages.Select(package => package.ID + " " + package.Version)), "Install", "Cancel"))
+                if (!EditorUtility.DisplayDialog(
+                        "Warning from NuGet importer",
+                        "Now the api compatibility level for this project is "
+                        + PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup)
+                                        .ToString()
+                        + ". But below packages are builded for .NETFramework. Do you install them?"
+                        + "\n\n"
+                        + string.Join("\n", warningPackages.Select(package => package.ID + " " + package.Version)),
+                        "Install",
+                        "Cancel"
+                    ))
                 {
                     return false;
                 }
@@ -284,11 +350,25 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                 {
                     isInstalled = PackageManager.InstalledCatalog.ContainsKey(installPackage.ID);
                 }
-                Catalog catalog = isInstalled ? PackageManager.InstalledCatalog[installPackage.ID] : await NuGet.GetCatalog(installPackage.ID);
-                Catalogentry catalogEntry = catalog.GetAllCatalogEntry().First(entry => entry.version == installPackage.Version);
+
+                Catalog catalog = isInstalled
+                    ? PackageManager.InstalledCatalog[installPackage.ID]
+                    : await NuGet.GetCatalog(installPackage.ID);
+                Catalogentry catalogEntry =
+                    catalog.GetAllCatalogEntry().First(entry => entry.version == installPackage.Version);
                 if (catalogEntry.requireLicenseAcceptance)
                 {
-                    var option = EditorUtility.DisplayDialogComplex("NuGet importer", catalogEntry.id + " " + catalogEntry.version + " need agree license.\nUrl : " + catalogEntry.licenseUrl, "Agree", "Cancel", "Go url");
+                    var option = EditorUtility.DisplayDialogComplex(
+                        "NuGet importer",
+                        catalogEntry.id
+                        + " "
+                        + catalogEntry.version
+                        + " need agree license.\nUrl : "
+                        + catalogEntry.licenseUrl,
+                        "Agree",
+                        "Cancel",
+                        "Go url"
+                    );
                     switch (option)
                     {
                         case 0:
@@ -297,16 +377,32 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
                             return false;
                         case 2:
                             Help.BrowseURL(catalogEntry.licenseUrl);
-                            if (!EditorUtility.DisplayDialog("NuGet importer", catalogEntry.id + " " + catalogEntry.version + " need agree license.\nUrl : " + catalogEntry.licenseUrl, "Agree", "Cancel"))
+                            if (!EditorUtility.DisplayDialog(
+                                    "NuGet importer",
+                                    catalogEntry.id
+                                    + " "
+                                    + catalogEntry.version
+                                    + " need agree license.\nUrl : "
+                                    + catalogEntry.licenseUrl,
+                                    "Agree",
+                                    "Cancel"
+                                ))
                             {
                                 return false;
                             }
+
                             break;
                     }
                 }
             }
 
-            return !nativePackages.Any() || EditorUtility.DisplayDialog("NuGet importer", "Native plugins were found in the modifying package. You need to restart the editor to modify packages.\n(The current project will be saved and modify packages will be resumed after a restart.)", "Restart", "Cancel");
+            return !nativePackages.Any()
+                   || EditorUtility.DisplayDialog(
+                       "NuGet importer",
+                       "Native plugins were found in the modifying package. You need to restart the editor to modify packages.\n(The current project will be saved and modify packages will be resumed after a restart.)",
+                       "Restart",
+                       "Cancel"
+                   );
         }
 
         /// <summary>
@@ -353,6 +449,9 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
         /// <para>Operation result</para>
         /// <para>操作結果。</para>
         /// </returns>
-        protected abstract Task<OperationResult> Operate(ReadOnlyControlledPackages controlledPackages, PackageManager.OperateLock operateLock);
+        protected abstract Task<OperationResult> Operate(
+            ReadOnlyControlledPackages controlledPackages,
+            PackageManager.OperateLock operateLock
+        );
     }
 }

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/OperatePackage.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/PackageOperation/OperatePackage.cs
@@ -248,12 +248,12 @@ namespace kumaS.NuGetImporter.Editor.PackageOperation
 
 #if UNITY_2021_2_OR_NEWER
             if(PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup) == ApiCompatibilityLevel.NET_Standard){
-                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_1.Contains(package.targetFramework));
+                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_1.Contains(package.TargetFramework)).ToArray();
             }
 #else
             if (PlayerSettings.GetApiCompatibilityLevel(EditorUserBuildSettings.selectedBuildTargetGroup) == ApiCompatibilityLevel.NET_Standard_2_0)
             {
-                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_0.Contains(package.TargetFramework)).ToList();
+                warningPackages = installPackages.Where(package => !FrameworkName.STANDARD2_0.Contains(package.TargetFramework)).ToArray();
             }
 #endif
             if (warningPackages.Any())

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/Setup/SetupPackage.cs
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/Setup/SetupPackage.cs
@@ -16,11 +16,12 @@ namespace kumaS.NuGetImporter.Editor.Setup
     {
         private static readonly List<ApiCompatibilityLevel> EnableApiLevel = new List<ApiCompatibilityLevel>
         {
+#if UNITY_2021_2_OR_NEWER
+            ApiCompatibilityLevel.NET_Unity_4_8,
+            ApiCompatibilityLevel.NET_Standard
+#else
             ApiCompatibilityLevel.NET_4_6,
             ApiCompatibilityLevel.NET_Standard_2_0
-#if UNITY_2021_2_OR_NEWER
-            ,ApiCompatibilityLevel.NET_Unity_4_8
-            ,ApiCompatibilityLevel.NET_Standard
 #endif
         };
 

--- a/NuGetImporterForUnity/Packages/NuGet Importer/Editor/kumaS.NuGetImporter.Editor.asmdef
+++ b/NuGetImporterForUnity/Packages/NuGet Importer/Editor/kumaS.NuGetImporter.Editor.asmdef
@@ -10,9 +10,6 @@
     "overrideReferences": false,
     "precompiledReferences": [],
     "autoReferenced": true,
-    "defineConstraints": [
-        "ZIP_AVAILABLE"
-    ],
     "versionDefines": [
         {
             "name": "com.unity.ide.vscode",

